### PR TITLE
Limit underscore suffix to non-public members

### PR DESCRIPTION
**Pull request by rhaschke**
Augments #1939 by enforcing the new rule by clang-tidy.
[Here](https://github.com/rhaschke/moveit2/actions/runs/4185935378) is a CI job running clang-tidy on all source files (not only the diff as usually done for PRs).

### CODE REVIEW

**Review by tylerjw — APPROVED** [review]
Easy approve! Thank you.

**Comment by rhaschke** [comment]
I was hoping you would wait for the [linked CI job](https://github.com/rhaschke/moveit2/actions/runs/4185935378) to finish before merging. Let's see whether clang-tidy finds more fixes... Sorry for not being clear enough initially.

**Comment by tylerjw** [comment]
> I was hoping you would wait for the [linked CI job](https://github.com/rhaschke/moveit2/actions/runs/4185935378) to finish before merging. Let's see whether clang-tidy finds more fixes... Sorry for not being clear enough initially.

Oh, I'm sorry. I through because CI passed here that this was fixed and this was just adding a check for future changes.

**Comment by tylerjw** [comment]
Oof, I see there are a bunch more of these changes to make. Maybe we should revert this change for now :(

**Comment by rhaschke** [comment]
I have gone through most of these changes in ROS1 already: https://github.com/ros-planning/moveit/pull/3317.
I will try to apply those changes here as well...

**Pull request merged**
